src/openrct2-ui/windows/TopToolbar.cpp

```diff
@@ -2597,20 +2612,36 @@ static money32 try_place_ghost_scenery(
             break;
         }
         case 4:
+        {
             // Banners
             // 6e2612
-            cost = game_do_command(
-                map_tile.x, parameter_1 | 0x69, map_tile.y, parameter_2, GAME_COMMAND_PLACE_BANNER, parameter_3, 0);
+            uint8_t direction = (parameter_2 >> 8) & 0xFF;
+            int32_t z = (parameter_2 & 0xFF) * 16;
+            CoordsXYZD loc{ map_tile.x, map_tile.y, z, direction };
+            auto primaryColour = gWindowSceneryPrimaryColour;
+            auto bannerType = (parameter_1 & 0xFF00) >> 8;
+            auto bannerIndex = create_new_banner(0);
```

Comment:
I'm a little worried about what if two clients are placing banners at the same time, this would collide